src/core/Service.js

```diff
@@ -85,46 +89,51 @@ export default class Service extends EventEmitter {
    * @param {advertiseCallback} callback - This works similarly to the callback for a C++ service and should take the following params
    */
   advertise(callback) {
-    if (this.isAdvertised || typeof callback !== 'function') {
-      return;
+    if (this.isAdvertised) {
+      throw new Error('Cannot advertise the same Service twice!');
     }
 
-    this._serviceCallback = callback;
-    this.ros.on(this.name, this._serviceResponse.bind(this));
+    // Store the new callback for removal during un-advertisement
+    this._serviceCallback = (rosbridgeRequest) => {
```

Comment:
In python I am really against adding class variables outside the constructor. I don't know how you feel about it in JS/TS.

Reply:
This class variable is declared in the constructor, just set to the actual callback here. I don't think TypeScript even lets you add class members that weren't declared in the constructor or in the class scope :)

This reminds me that I should go around and clean up all the places where class fields are being initialized with constants (not constructor parameters) and move them out of the constructor into the body of the class - I'll do that here first.